Use libc++ on Mac OS

Trying to cross-compile for Mac OS on osxcross.
A standard header is not found. It's because in old SDKs of Mac OS the default STL is GCC's. It's not updated by Apple, and old.

It may be a good idea to permit builds with libc++ (or make it default?). Old Mac systems like Snow Leopard don't come with libc++, but Macports allows to acquire it.

```
/home/jpc/Documents/Projects/ADLplug/thirdparty/libADLMIDI/src/adlmidi_private.hpp:90:10: fatal error: 
      'cinttypes' file not found
#include <cinttypes> //PRId32, PRIu32, etc.
```

You can enable libc++ with these CMake lines.

```
if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
endif()
```

EDIT: not the good cmake code, fix

--- a/src/adlmidi_private.hpp
+++ b/src/adlmidi_private.hpp
@@ -87,7 +87,11 @@ typedef int32_t ssize_t;
 #include <cmath>
 #include <cstdarg>
 #include <cstdio>
+#if !(defined(__APPLE__) && defined(__GLIBCXX__))
 #include <cinttypes> //PRId32, PRIu32, etc.
+#else
+#include <inttypes.h>
+#endif
 #include <vector> // vector
 #include <deque>  // deque
 #include <cmath>  // exp, log, ceil

